hbase-client/src/test/java/org/apache/hadoop/hbase/util/TestRoundRobinPoolMap.java

```diff
@@ -45,58 +46,53 @@ protected PoolType getPoolType() {
   }
 
   @Test
-  public void testSingleThreadedClient() throws InterruptedException, ExecutionException {
-    Random rand = ThreadLocalRandom.current();
-    String randomKey = String.valueOf(rand.nextInt());
-    String randomValue = String.valueOf(rand.nextInt());
-    // As long as the pool is not full, we'll get null back.
-    // This forces the user to create new values that can be used to populate
-    // the pool.
-    runThread(randomKey, randomValue, null);
-    assertEquals(1, poolMap.size(randomKey));
+  public void testSingleThreadedClient() throws InterruptedException {
+    String key = "key";
+    String value = "value";
+    // As long as the pool is not full, get calls the supplier
+    runThread(key, () -> value, value);
+    assertEquals(1, poolMap.values().size());
   }
 
   @Test
-  public void testMultiThreadedClients() throws InterruptedException, ExecutionException {
-    Random rand = ThreadLocalRandom.current();
-    for (int i = 0; i < POOL_SIZE; i++) {
-      String randomKey = String.valueOf(rand.nextInt());
-      String randomValue = String.valueOf(rand.nextInt());
-      // As long as the pool is not full, we'll get null back
-      runThread(randomKey, randomValue, null);
-      // As long as we use distinct keys, each pool will have one value
-      assertEquals(1, poolMap.size(randomKey));
+  public void testMultiThreadedClients() throws InterruptedException {
+    for (int i = 0; i < KEY_COUNT; i++) {
+      String key = Integer.toString(i);
+      String value = Integer.toString(2 * i);
+      // As long as the pool is not full, we'll get the supplied value back
+      runThread(key, () -> value, value);
     }
+
+    assertEquals(KEY_COUNT, poolMap.values().size());
```

Comment:
we should drop the "as long as the pool is not full" comment here. This first stanza is testing "as long as we use distinct keys we should get get back the supplied value" and "we end up with 1 value per pool".